prime-router/src/main/kotlin/azure/ApiKeysFunctions.kt

```diff
@@ -48,43 +116,128 @@ class ApiKeysFunctions(private val settingsFacade: SettingsFacade = SettingsFaca
+            ApiResponse(responseCode = HTTP_400_BAD_REQ, description = HTTP_400_ERR_MSG) // NOSONAR
+        ],
+    )
+    @GET
+    @Path("settings/organizations/{organizationName}/public-keys")
     fun get(
         @HttpTrigger(
             name = "getApiKeys",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
             route = "settings/organizations/{organizationName}/public-keys"
-        ) request: HttpRequestMessage<String?>,
-        @BindingName("organizationName") orgName: String
+        ) @Parameter(hidden = true) request: HttpRequestMessage<String?>,
+        @Parameter(
+            name = ORG_NAME_STR,
+            description = ORG_NAME_DESC
+        )
+        @PathParam(ORG_NAME_STR)
+        @BindingName(ORG_NAME_STR) orgName: String
     ): HttpResponseMessage {
         return getApiKeysForOrg(request, orgName)
     }
 
     @FunctionName("getApiKeysV1")
+    @SuppressWarnings("all")
+    @Operation(
+        summary = "Retrieve API keys for the organization (v1), return API keys when successful",
+        description = OPERATION_GET_KEYS_DESC, // NOSONAR
+        tags = [KEY_MGMT_TAG], // NOSONAR
+        parameters = [
+            Parameter(
+                name = ORG_NAME_STR, // NOSONAR
+                required = true, // NOSONAR
+                description = ORG_NAME_DESC // NOSONAR
+            )
+        ],
+        responses = [
+            ApiResponse(
+                responseCode = HTTP_200_OK, // NOSONAR
+                description = HTTP_200_GET_MSG, // NOSONAR
+                content = [
+                    Content( // NOSONAR
+                        mediaType = "application/json", // NOSONAR
+                        schema = Schema(implementation = ApiKeysResponse::class) // NOSONAR
+                    )
+                ]
+            ),
+            ApiResponse(responseCode = HTTP_404_NOT_FOUND, description = HTTP_404_ERR_MSG), // NOSONAR
+            ApiResponse(responseCode = HTTP_400_BAD_REQ, description = HTTP_400_ERR_MSG) // NOSONAR
+        ]
+    )
+    @GET
+    @Path("v1/settings/organizations/{organizationName}/public-keys")
     fun getV1(
         @HttpTrigger(
             name = "getApiKeysV1",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
             route = "v1/settings/organizations/{organizationName}/public-keys"
-        ) request: HttpRequestMessage<String?>,
-        @BindingName("organizationName") orgName: String
+        ) @Parameter(hidden = true) request: HttpRequestMessage<String?>,
+        @Parameter(
+            name = ORG_NAME_STR,
+            description = ORG_NAME_DESC
+        )
+        @PathParam(ORG_NAME_STR)
+        @BindingName(ORG_NAME_STR) orgName: String
     ): HttpResponseMessage {
         return getApiKeysForOrg(request, orgName, true)
     }
 
     @FunctionName("postApiKey")
+    @POST
```

Comment:
In the swagger UI, the "Example" value for Request Body is not not very helpful and is incorrect (it points to `HTTPRequestMessageString`), I think we should update it. Also why I think we should actually try to use the docs to submit a request, it will help us catch things like this.

Reply:
> HTTPRequestMessageString

that is part of the post function signature and it worked as is (was like that on master branch), and it is working as is, not sure if change that will also re-factor references in the function body?

Reply:
> In the swagger UI, the "Example" value for Request Body is not not very helpful and is incorrect (it points to `HTTPRequestMessageString`), I think we should update it. Also why I think we should actually try to use the docs to submit a request, it will help us catch things like this.

the actual payload of request and responses could be captured from a local setup ...., agree that the tool derived ones are not good, I might need hand craft them (example json, and it's json schema).

Deploying to staging works too, but I am not sure I have been given azure account yet....

Will do something based on local setup.

Reply:
@arnejduranovic I looked at the azure func and noticed that HttpRequestMessage<String?> is all over the place, see code excerpted below, customizing each end points is a lot of work, not sure if we want to do it?:

@FunctionName("getsendfailures")
    fun getSendFailures(
        @HttpTrigger(
            name = "getsendfailures",
            methods = [HttpMethod.GET],
            authLevel = AuthorizationLevel.ANONYMOUS,
            route = "adm/getsendfailures" // this can NOT be "admin/" or it fails.
        )
        request: HttpRequestMessage<String?>,
    ): HttpResponseMessage {

Reply:
FYI, for GET operations, the request body is suppose to be empty, we have to put @Parameter(hidden = true) request: HttpRequestMessage<String?> to suppress swagger editor validation error...

For post, a good sample + schema for the request body might be value to the API docs.